### PR TITLE
Story 15

### PR DESCRIPTION
**Pull request by amhursh**
Related Issue: #13 

Summary: Added user checkout functionality. cart contents can now be fed into a create order action and said orders are displayed with status in the orders index.

Author(s): amhursh

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -8,4 +8,16 @@ def show
 		@items = @order.items.distinct
 	end
 
+	def create
+		order = Order.new(user_id: current_user.id)
+		order.add_items(@cart)
+		if order.save
+			flash[:success] = "Order was successfully placed"
```
Comment (limsammy) [inline app/controllers/orders_controller.rb:15]: Glad we're adding flash messages now

File: app/models/order.rb [hunk 2]
```diff
@@ -19,4 +19,11 @@ def item_subtotal(item)
   	items.where(id: id).sum(:price)
   end
 
+  def add_items(cart)
+    cart.contents.each do |item_id, quantity|
+      item = Item.find(item_id)
+      quantity.times {items << item}
```
Comment (limsammy) [inline app/models/order.rb:25]: this is awesome.

File: app/views/carts/show.html.erb [hunk 3]
```diff
@@ -36,7 +36,7 @@
     <h3 class="h-tag-no-top-margin-title center-text">Cart Total: $<%= @cart.total %></h3>
     <div class="cart_checkout center-text">
       <% if current_user %>
-      <%= link_to "Checkout", cart_path %>
+      <%= link_to "Checkout", orders_path, method: :post %>
```
Comment (limsammy) [inline app/views/carts/show.html.erb:39]: Nice catch

**Review by limsammy — COMMENTED** [review]
Looks good to me, waiting on other reviewers.

File: app/views/orders/index.html.erb [hunk 4]
```diff
@@ -3,8 +3,9 @@
 <ul>
 <% current_user.orders.each do |order| %>
 	<li>
-		Order id:<%= order.id %><br>
-		<%= order.created_at %>
+		Order id: <%= order.id %><br>
```
Comment (randy-darsh) [inline app/views/orders/index.html.erb:6]: loving the indentation aaron

File: app/views/carts/show.html.erb (hunk 3 above)
Comment (randy-darsh) [reply app/views/carts/show.html.erb:39]: i agree

**Comment by randy-darsh** [comment]
MERGING... ENGAGE!

**Pull request merged**